File: src/main/java/org/example/TC/명예의전당.java
package org.example.TC;

import java.util.PriorityQueue;
/*https://school.programmers.co.kr/learn/courses/30/lessons/138477?language=java*/
public class 명예의전당 {
    public int[] solution(int k, int[] score) {
        PriorityQueue<Integer>priorityQueue = new PriorityQueue<>();
        int[] answer = new int[score.length];

        for(int i = 0; i < score.length; i++) {
            priorityQueue.add(score[i]);
            if(priorityQueue.size()>k) {
                priorityQueue.poll();
            }
            /*peek()은 스택 위의 가장 위의 element 를 반환한다.*/
            answer[i] = priorityQueue.peek();
        }
        return answer;
    }
}

File: src/main/java/org/example/TCK/명예의전당.kt
package org.example.TCK

import java.util.PriorityQueue

class 명예의전당 {
    fun solution(k:Int, score:IntArray):IntArray{
        var answer: MutableList<Int> = mutableListOf<Int>()
        val list = PriorityQueue<Int>()
        for(i:Int in score){
            if(list.size<k) {
                list.add(i);
            } else {
                if(list.peek() < i){
                    list.add(i)
                    list.poll()
                }
            }
            answer.add(list.peek())
        }
        return answer.toIntArray()
    }
}
